privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/PrivacyDashboardHybridViewModel.kt

```diff
@@ -48,37 +48,72 @@ import javax.inject.Inject
 @ContributesViewModel(ActivityScope::class)
 class PrivacyDashboardHybridViewModel @Inject constructor(
     private val userWhitelistDao: UserWhitelistDao,
-    private val contentBlocking: ContentBlocking,
     private val pixel: Pixel,
     private val dispatcher: DispatcherProvider,
     @AppCoroutineScope private val appCoroutineScope: CoroutineScope,
-    private val siteProtectionsViewStateMapper: SiteProtectionsViewStateMapper
+    private val siteViewStateMapper: SiteViewStateMapper,
+    private val requestDataViewStateMapper: RequestDataViewStateMapper,
+    private val protectionStatusViewStateMapper: ProtectionStatusViewStateMapper,
+    private val privacyDashboardPayloadAdapter: PrivacyDashboardPayloadAdapter
 ) : ViewModel() {
 
     private val command = Channel<Command>(1, DROP_OLDEST)
 
     sealed class Command {
         class LaunchReportBrokenSite(val data: BrokenSiteData) : Command()
+        class OpenURL(val url: String) : Command()
     }
 
     data class ViewState(
-        val userSettingsViewState: UserSettingsViewState,
-        val siteProtectionsViewState: SiteProtectionsViewState,
-        val userChangedValues: Boolean = false
+        val siteViewState: SiteViewState,
+        val userChangedValues: Boolean = false,
+        val requestData: RequestDataViewState,
+        val protectionStatus: ProtectionStatusViewState
     )
 
-    data class UserSettingsViewState(
-        val privacyProtectionEnabled: Boolean
+    data class ProtectionStatusViewState(
+        val allowlisted: Boolean,
+        val denylisted: Boolean,
+        val enabledFeatures: List<String>,
+        val unprotectedTemporary: Boolean
+    )
+
+    data class RequestDataViewState(
+        val installedSurrogates: List<String>? = null,
+        val requests: List<DetectedRequest>
+    )
+
+    data class DetectedRequest(
+        val category: String?,
+        val eTLDplus1: String?,
+        val entityName: String,
+        val ownerName: String?,
+        val pageUrl: String,
+        val prevalence: Double,
+        val state: State,
+        val url: String
     )
 
-    data class SiteProtectionsViewState(
+    sealed class State {
+        data class Blocked(val blocked: Any = Any()) : State()
+        data class Allowed(val allowed: Reason) : State()
```

Comment:
The val should probably be named `reason` for consistency, `allow` doesn't really make a lot of sense.

Reply:
this is just to mimic the values on the UI ([so the privacy dashboard](https://duckduckgo.github.io/privacy-dashboard/example/docs/interfaces/Generated_Schema_Definitions.DetectedRequest.html#state)). I see it can lead to confusion and write a comment in the ViewModel.